src/Common/Notifications/Controller.php

```diff
@@ -91,6 +91,16 @@ public function register_ian() {
 				],
 			]
 		);
+
+		/**
+		 * Allow plugins to hook in and add themselves,
+		 * running their own actions after IAN is initiated.
+		 *
+		 * @since 6.4.0
+		 *
+		 * @param self $ian The IAN instance.
+		 */
+		do_action( 'tec_common_ian_loaded', $this );
```

Comment:
Define a static property instead in the controller e.g.

`public static string $registration_action = 'tec_common_ian_loaded';`

The container will fire that action on its own after the controller has been registered